lib/IAMClient.js

```diff
@@ -284,40 +284,76 @@ class VaultClient {
     /**
      * Update Quota of an account
      *
-     * @param {string} accountName - account name
      * @param {number} quota - maximum quota for the account
+     * @param {object} options - additional creation params
+     * @param {string} [options.accountName] - account name
      * @param {VaultClient~requestCallback} callback - callback
      * @returns {undefined}
      */
-    updateAccountQuota(accountName, quota, callback) {
-        const conv = Number.isNaN(Number.parseInt(quota, 10));
-        assert(typeof accountName === 'string' && accountName !== '',
-            'accountName is required');
-        assert(typeof quota === 'number' && !conv
-            && quota >= 0, 'Quota must be a non-negative number');
-        this.request('POST', '/', true, callback, {
+    updateAccountQuota(quota, options, callback) {
+        const { accountName } = options || {};
+        const data = {
             Action: 'UpdateAccountQuota',
             Version: '2010-05-08',
-            AccountName: accountName,
             quotaMax: quota,
-        });
+        };
+
+        const quotaIsValid = typeof quota === 'number' && !Number.isNaN(Number.parseInt(quota, 10)) && quota > 0;
+        assert(quotaIsValid, 'Quota must be a strictly positive number');
+        if (accountName) {
+            assert(typeof accountName === 'string', 'the account name, if set, should be a string');
+            data.AccountName = accountName;
+        }
+        this.request('POST', '/', true, callback, data);
     }
 
     /**
      * Delete Quota of an account
      *
-     * @param {string} accountName - account name
+     * @param {object} options - additional creation params
+     * @param {string} [options.accountName] - account name
```

Comment:
this will break compatibility:
* is it really acceptable?
* if we break compatibility, we should create a new major version. Otherwise, we need to maintain compatibility (possibly supporting both accountName and options.accountName)
* do we expect to have more parameters? if not, why change the api at all...

Reply:
we're no longer breaking the compatibility see commit : https://github.com/scality/vaultclient/pull/442/commits/bedefa240fb8f76860557af58533008a8b8c0bb1#diff-6ed3f39879bec9944e1a6e9d0d3c1bcbc7209a7eab826e5c3819048e4841163d